cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -182,7 +182,8 @@ func (gc *GceCache) GetMigForInstance(instanceRef GceRef) (Mig, error) {
 
 			migRef, found := gc.instanceRefToMigRef[instanceRef]
 			if !found {
-				return nil, fmt.Errorf("instance %+v belongs to unknown mig", instanceRef)
+				klog.Warningf("instance %+v belongs to unknown mig", instanceRef)
+				return nil, nil
```

Comment:
Please also add the `gc.instancesFromUnknownMigs map[GceRef]bool{}` and put the entry the map here.
At the top of the function in `else` branch of loop from line `154` add checking of this new map. If entry is found return `(nil, nil)` from function immediately.

Also add code for cleaning up the new map together with `gc.instanceRefToMigRef`.

Without the new `gc.instancesFromUnknownMigs` map we will be listing instances of mig, which appears to hold the problematic instance, every loop. That will harm the performance.

Reply:
 @losipiuk I've done the requests changes 👍 
I've done a `map[GceRef]struct{}` which is lighter than a bool one as we just store it as index/set. Let me know if everything's fine